lib/ssl-test.rb

```diff
@@ -56,4 +63,88 @@ def self.matching_domains domains, hostname
     domains.map {|s| Regexp.new("\A#{Regexp.escape(s).gsub('\*', '[^.]+')}\z") }
       .select {|domain| domain.match?(hostname) }
   end
+
+  # https://docs.ruby-lang.org/en/2.2.0/OpenSSL/OCSP.html
+  # https://stackoverflow.com/questions/16244084/how-to-programmatically-check-if-a-certificate-has-been-revoked#answer-16257470
+  def self.test_ocsp_revocation cert, chain, open_timeout: 5, read_timeout: 5, redirection_limit: 5
+    issuer = chain.last
+
+    digest = OpenSSL::Digest::SHA1.new
+    certificate_id = OpenSSL::OCSP::CertificateId.new(cert, issuer, digest)
+
+    request = OpenSSL::OCSP::Request.new
+    request.add_certid certificate_id
+    request.add_nonce
+
+    authority_info_access = cert.extensions.find do |extension|
+      extension.oid == "authorityInfoAccess"
+    end
+
+    descriptions = authority_info_access.value.split("\n")
+    ocsp = descriptions.find do |description|
+      description.start_with?("OCSP")
+    end
+
+    ocsp_uri = URI(ocsp[/URI:(.*)/, 1])
+    http_response = follow_ocsp_redirects(ocsp_uri, request.to_der, open_timeout: open_timeout, read_timeout: read_timeout, redirection_limit: redirection_limit)
```

Comment:
In [the doc example](https://docs.ruby-lang.org/en/2.2.0/OpenSSL/OCSP.html) it states:
> Note that we only handle HTTP requests and don't handle any redirects in this example, so this is insufficient for serious use.

which is why I added `follow_ocsp_redirects` to follow the redirections (up to `redirection_limit` times).

Reply:
:+1: